x-pack/plugins/ml/public/alerting/anomaly_detection_alerts_table/alert_actions.tsx

```diff
@@ -0,0 +1,210 @@
+import {
+  ALERT_RULE_NAME,
+  ALERT_RULE_UUID,
+  ALERT_STATUS,
+  ALERT_STATUS_ACTIVE,
+  ALERT_UUID,
+} from '@kbn/rule-data-utils';
+import { useBulkUntrackAlerts } from '@kbn/triggers-actions-ui-plugin/public';
+import { type Alert } from '@kbn/triggers-actions-ui-plugin/public/types';
+import { PLUGIN_ID } from '../../../common/constants/app';
+import { useMlKibana } from '../../application/contexts/kibana';
+
+export interface AlertActionsProps {
+  alert: Alert;
+  ecsData: Ecs;
+  id?: string;
+  refresh: () => void;
+  setFlyoutAlert: React.Dispatch<React.SetStateAction<any | undefined>>;
+}
+
+export function AlertActions({
+  alert,
+  ecsData,
+  id: pageId,
+  refresh,
+  setFlyoutAlert,
+}: AlertActionsProps) {
+  const {
+    cases,
+    http: {
+      basePath: { prepend },
+    },
+  } = useMlKibana().services;
+  const casesPrivileges = cases?.helpers.canUseCases();
+
+  const { mutateAsync: untrackAlerts } = useBulkUntrackAlerts();
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+
+  const ruleId = alert[ALERT_RULE_UUID]?.[0] ?? null;
+  const alertId = alert[ALERT_UUID]?.[0] ?? '';
+
+  const linkToRule = ruleId
+    ? prepend(`/app/management/insightsAndAlerting/triggersActions/rule/${ruleId}`)
+    : null;
+
+  const caseAttachments: CaseAttachmentsWithoutOwner = useMemo(() => {
+    return ecsData?._id
+      ? [
+          {
+            alertId: alertId ?? '',
+            index: ecsData?._index ?? '',
+            type: AttachmentType.alert,
+            rule: {
+              id: ruleId,
+              name: alert[ALERT_RULE_NAME]![0],
+            },
+            owner: PLUGIN_ID,
+          },
+        ]
+      : [];
+  }, [alert, alertId, ecsData?._id, ecsData?._index, ruleId]);
+
+  const isActiveAlert = useMemo(() => alert[ALERT_STATUS]![0] === ALERT_STATUS_ACTIVE, [alert]);
+
+  const onSuccess = useCallback(() => {
+    refresh();
+  }, [refresh]);
+
+  const createCaseFlyout = cases!.hooks.useCasesAddToNewCaseFlyout({ onSuccess });
+  const selectCaseModal = cases!.hooks.useCasesAddToExistingCaseModal({ onSuccess });
+
+  const closeActionsPopover = () => {
+    setIsPopoverOpen(false);
+  };
+
+  const toggleActionsPopover = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+  };
+
+  const handleAddToNewCaseClick = () => {
+    createCaseFlyout.open({ attachments: caseAttachments });
+    closeActionsPopover();
+  };
+
+  const handleAddToExistingCaseClick = () => {
+    selectCaseModal.open({ getAttachments: () => caseAttachments });
+    closeActionsPopover();
+  };
+
+  const handleUntrackAlert = useCallback(async () => {
+    await untrackAlerts({
+      indices: [ecsData?._index ?? ''],
+      alertUuids: [alertId],
+    });
+    onSuccess();
+  }, [untrackAlerts, alertId, ecsData, onSuccess]);
+
+  const actionsMenuItems = [
+    ...(casesPrivileges?.create && casesPrivileges.read
+      ? [
+          <EuiContextMenuItem
+            data-test-subj="add-to-existing-case-action"
```

Comment:
nit: Shall we update dataTestSubj to match normal convention (e.g. addToExistingCaseAction)? Unless this is due to a testing service.